app/src/pages/trace/SpanAnnotationActionMenu.tsx

```diff
@@ -38,15 +41,28 @@ export function SpanAnnotationActionMenu(props: SpanAnnotationActionMenuProps) {
     onSpanAnnotationActionError,
   } = props;
   const [confirmDialog, setConfirmDialog] = useState<ReactNode>(null);
+  const { projectId } = useParams();
+  const { timeRange } = useLastNTimeRange();
   const [commitDelete, isCommittingDelete] =
     useMutation<SpanAnnotationActionMenuDeleteMutation>(graphql`
       mutation SpanAnnotationActionMenuDeleteMutation(
         $annotationId: GlobalID!
         $spanId: GlobalID!
+        $annotationName: String!
+        $projectId: GlobalID!
+        $timeRange: TimeRange!
       ) {
         deleteSpanAnnotations(input: { annotationIds: [$annotationId] }) {
           query {
-            node(id: $spanId) {
+            project: node(id: $projectId) {
+              ...ProjectPageHeader_stats
```

Comment:
same as above but in reverse - refetch stats to potentially remove the only instance of an annotation using a particular name

refetch ValueFragment to update the stats associated with a particular annotation name